OpenRA.Mods.Common/Traits/World/MusicPlaylist.cs

```diff
@@ -162,13 +162,22 @@ void Play()
 			if (!SongExists(currentSong) || (CurrentSongIsBackground && IsBackgroundMusicMuted))
 				return;
 
-			Game.Sound.PlayMusicThen(currentSong, () =>
+			if (Game.Settings.Sound.Repeat)
 			{
-				if (!CurrentSongIsBackground && !Game.Settings.Sound.Repeat)
-					currentSong = GetNextSong();
+				Game.Sound.PlayMusic(currentSong, looped: true);
+				Game.Sound.SetAfterCompletion(PlayNextSong);
```

Comment:
Stupid question: Can we not just add a `bool looped` to `PlayMusicThen` now instead?

Reply:
Good question. I don't even need another boolean as I have it statically available, which eliminates even more code paths.